Added a typealias subsection and regroup other type related subsectio…

paulmiard opened the pull request:
…ns into the existing 'Types' section

paulmiard committed:
Added a typealias subsection and regroup other type related subsections into the existing 'Types' section

Haud commented on README.md:
```diff
@@ -16,8 +16,6 @@
 	* [Access Modifiers](#access-modifiers)
 	* [Imports](#imports)
-	* [Type Declarations](#type-declarations)
-	* [Type Inference](#type-inference)
```
why did you delete these?

ThibaultKlein replied:
because everything is under `Type` now

paulmiard replied:
Yeah they are now subsections of type so I figured they'd be better off the toc, otherwise we need one more tab

ThibaultKlein commented on README.md:
```diff
@@ -150,10 +148,68 @@ let scaleString = scale.stringValue
+typealias AgeAndNameProcessor = (Integer, String) -> Void
+
+var intValue: Int
+
```
you can remove these spaces

paulmiard replied:
done

ThibaultKlein commented on README.md:
```diff
@@ -150,10 +148,68 @@ let scaleString = scale.stringValue
+typealias ClosureType = (ParameterTypes) -> (ReturnType)
+typealias AgeAndNameProcessor = (Integer, String) -> Void
+
+var intValue: Int
```
I would put a class or a struct declaration as an example instead, so it's clear that `typealias` is outside of a type scope

paulmiard replied:
done. Also commented the first typealias example as it would not compile. Keeping it as a reference

paulmiard committed:
Addressed PR comments

ThibaultKlein commented:
👍

Haud commented:
What about typealias as part of protocol conformance? then it has to be part of the class

paulmiard commented on README.md:
````diff
@@ -150,10 +148,73 @@ let scaleString = scale.stringValue
+	}
+}
+```
+
````
re:Chris comment: `If declaring a typealias for protocol conformance, it should be declared at the top of the class, before anything else`

paulmiard replied:
Done

Haud commented on README.md:
```diff
@@ -150,10 +148,101 @@ let scaleString = scale.stringValue
+
+#### typealias ####
+
+Typealias declaration should precede any other type declaration.
```
`Typealias declarations`

Haud commented:
Small grammatical note, but otherwise 👍  for me once resolved

paulmiard commented:
@ThibaultKlein I double checked it looks good to merge. I let you check as well if you want, or just click the merge button.

ThibaultKlein commented on README.md:
```diff
@@ -150,10 +148,101 @@ let scaleString = scale.stringValue
+#### Type Declarations ####
+
+When declaring types, the colon should be placed immediately after the identifier followed by one space
+and the type name.
```
remove the new line here

paulmiard replied:
done (even though it doesn't matter it's markdown)

ThibaultKlein commented on README.md:
````diff
@@ -150,10 +148,101 @@ let scaleString = scale.stringValue
+
+```swift
+protocol Configurable {
+    associatedtype InputData
````
add a new line after `InputData`

paulmiard replied:
done

ThibaultKlein commented:
All your examples are missing the space between the first and last curly brackets (see Bracket Syntax section)

ThibaultKlein commented on README.md:
```diff
@@ -150,10 +148,101 @@ let scaleString = scale.stringValue
+
+*Rationale* The compiler is pretty smart, and we should utilize it where necessary. It is generally obvious what the
+type is going to be in the instances above, so unless we need to be more explicit (as in the last examples above),
+it is better to omit unneeded words.
```
remove the new line

paulmiard replied:
done (even though it doesn't matter it's markdown)

ThibaultKlein commented on README.md:
```diff
@@ -422,7 +490,7 @@ of how they wish to handle the result of this failable operation.
 
-In general, you should avoid `NSError` in Swift in favor of defining your own `ErrorType`. However, in the event you do need to use `NSError` 
+In general, you should avoid `NSError` in Swift in favor of defining your own `ErrorType`. However, in the event you do need to use `NSError`
 (for interop with Objective-C, for instance):
```
remove new line

paulmiard replied:
done (even though it doesn't matter it's markdown)

paulmiard commented:
@ThibaultKlein I addressed your comments then rebased and fixed conflicts. Let me know. If it looks good to you please merge. 

Also Atom removed a few extra spaces here and there for us..
